### PR TITLE
Use pydantic settings

### LINKED ISSUE #82: Use pydantic settings class for configuration
**Describe the new feature you would like to see**
Use the pydantic models for a settings class instead of the current implementation of `config` in order to keep configuration variables.

**Describe the solution you'd like**
This solution will enable overwriting all configuration variables directly through setting environment variables which is very useful when using docker and docker-compose.

**How will this feature affect the current Threat Model?**
Should not affect the threat model.

### PR DESCRIPTION
**Pull request by aestene**
Closes #82 

The launch now also created an external docker network which the ISAR container is able to connect with.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,3 +17,8 @@ services:
       - /tmp/.X11-unix:/tmp/.X11-unix
     runtime: nvidia
     tty: true
+    networks:
+      - proxynet
+networks:
+    proxynet:
+      name: turtlenet
```
Comment (nicholasdalhaug) [inline docker-compose.yml:24]: Why this is used and the relevance to the `host = noetic` in `default.ini` is not obvious.

File: setup.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
     packages=find_packages(where="src"),
     package_dir={"": "src"},
     package_data={
-        "isar_turtlebot": ["config/default.ini", "config/maps/turtleworld.json"]
+        "isar_turtlebot": ["settings/settings.env", "config/maps/turtleworld.json"]
```
Comment (nicholasdalhaug) [inline setup.py:15]: This file is empty and the readme is not updated with how it is used, right?

**Review by nicholasdalhaug — CHANGES_REQUESTED** [review]
Looks pretty good, I think, but just a few questions.

File: docker-compose.yml (hunk 1 above)
Comment (aestene) [reply docker-compose.yml:24]: As for noetic as host the entire default.ini file was removed in the next commit so it was a setting that was leftover and should have been set back to localhost in that commit.

Comment (aestene) [reply docker-compose.yml:24]: The reason for setting up the network is to have a specific docker network running for the simulation. This enables different docker containers to be started which may connect to the docker container that is running the simulation.

So if we want to start ISAR in a docker container we would be able to connect it to the simulation by saying it should be on an external network named `turtlenet`

File: setup.py (hunk 2 above)
Comment (aestene) [reply setup.py:15]: Yes, this line should be removed. I will have a look at the readme.

File: docker-compose.yml (hunk 1 above)
Comment (nicholasdalhaug) [reply docker-compose.yml:24]: OK. Are you sure this works across docker-compose files? That would assume we at least run both on the same machine, right?

Comment (aestene) [reply docker-compose.yml:24]: Yes, this works across docker-compose files. I have only tested on the same machine but I think you are correct in that statement.

Comment (aestene) [reply docker-compose.yml:24]: Some adjustments might have to be made once we actually deploy this if we decide to go for multiple VMs in that environment.

File: setup.py (hunk 2 above)
Comment (aestene) [reply setup.py]: As for the readme there is no reference to the config that should have changed as far as I can tell.

File: src/isar_turtlebot/config/default.ini [hunk 3]
```diff
@@ -5,7 +5,7 @@ image_filetype = jpg
 thermal_image_filetype = png
 
 [ros_bridge]
-host = localhost
+host = noetic
```
Comment (nicholasdalhaug) [inline src/isar_turtlebot/config/default.ini]: Should we remove this?

File: docker-compose.yml (hunk 1 above)
Comment (nicholasdalhaug) [reply docker-compose.yml:24]: I suggest we at least rename it to `turtlenet`, I do not see why we have two names for it.

**Review by nicholasdalhaug — CHANGES_REQUESTED** [review]

File: src/isar_turtlebot/config/default.ini (hunk 3 above)
Comment (aestene) [reply src/isar_turtlebot/config/default.ini]: It has been removed. Unless you mean removing it for that specific commit?

Comment (nicholasdalhaug) [reply src/isar_turtlebot/config/default.ini]: I mean for that commit, yes.

Comment (aestene) [reply src/isar_turtlebot/config/default.ini]: Should be correct now.

**Review by nicholasdalhaug — APPROVED** [review]

**Pull request merged**